frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,243 @@
+        );
+      }
+      if (label.length > 63) {
+        errors.push(`**${label}** can't exceed 63 characters`);
+      }
+    });
+
+    return errors;
+  };
+
+  const handleEditComplete = (
+    _event: MouseEvent | KeyboardEvent,
+    newText: string,
+    currentLabel?: string,
+  ) => {
+    if (!newText) {
       return;
     }
-    const copy = [...unsavedLabels];
-    copy[index] = newText;
-    setUnsavedLabels(copy);
+
+    setUnsavedLabels((prev) => {
+      if (currentLabel) {
+        const index = prev.indexOf(currentLabel);
+        if (index === -1) {
+          return [...prev, newText];
+        }
+
+        const newLabels = [...prev];
+        newLabels[index] = newText;
+        return newLabels;
+      }
+      return [...prev, newText];
+    });
   };
+
   const removeUnsavedLabel = (text: string) => {
     if (isSavingEdits) {
       return;
     }
     setUnsavedLabels(unsavedLabels.filter((label) => label !== text));
   };
-  const addUnsavedLabel = (text: string) => {
+
+  const addNewLabel = () => {
     if (isSavingEdits) {
       return;
     }
-    setUnsavedLabels([...unsavedLabels, text]);
-  };
+    const baseLabel = 'New Label';
+    let counter = 1;
+    let newLabel = baseLabel;
 
-  // Don't allow a label that matches a non-label property key or another label (as they stand before saving)
-  // Note that this means if you remove a label and add it back before saving, that is valid
-  const reservedKeys = [
-    ...allExistingKeys.filter((key) => !labels.includes(key)),
-    ...unsavedLabels,
-  ];
-
-  const [isAddLabelModalOpen, setIsAddLabelModalOpen] = React.useState(false);
-  const [addLabelInputValue, setAddLabelInputValue] = React.useState('');
-  const addLabelInputRef = React.useRef<HTMLInputElement>(null);
-  let addLabelValidationError: string | null = null;
-  if (reservedKeys.includes(addLabelInputValue)) {
-    addLabelValidationError = 'Label must not match an existing label or property key';
-  } else if (addLabelInputValue.length > 63) {
-    addLabelValidationError = "Label text can't exceed 63 characters";
-  }
-
-  const toggleAddLabelModal = () => {
-    setAddLabelInputValue('');
-    setIsAddLabelModalOpen(!isAddLabelModalOpen);
-  };
-  React.useEffect(() => {
-    if (isAddLabelModalOpen && addLabelInputRef.current) {
-      addLabelInputRef.current.focus();
+    while (unsavedLabels.includes(newLabel)) {
+      newLabel = `${baseLabel} ${counter}`;
+      counter++;
     }
-  }, [isAddLabelModalOpen]);
-
-  const addLabelModalSubmitDisabled = !addLabelInputValue || !!addLabelValidationError;
-  const submitAddLabelModal = (event?: React.FormEvent) => {
-    event?.preventDefault();
-    if (!addLabelModalSubmitDisabled) {
-      addUnsavedLabel(addLabelInputValue);
-      toggleAddLabelModal();
+
+    setUnsavedLabels((prev) => [...prev, newLabel]);
+  };
+
+  const labelErrors = validateLabels(unsavedLabels);
```

Comment:
`validateLabels` doesn't really need to take `unsavedLabels` as an argument at all, it can just directly use `unsavedLabels` in its body instead of its `labelsList`.